test/integration/model/sync.test.js

```diff
@@ -403,6 +405,33 @@ describe(Support.getTestDialectTeaser('Model'), () => {
           expect(results.filter(r => r.name === 'wow_my_index')).to.have.length(1);
         }
       });
+
+      it('should remove old foreign keys when sync with {alter: true} if custom schema', async function () {
+        if (current.dialect.supports.schemas) {
+          let schema = 'test';
+          if (['mysql', 'mariadb'].includes(current.dialect.name)) {
```

Comment:
`mariadb` supports `createSchema`

Reply:
Yes. But I need some time to unravel and understand the logic of ```getForeignKeyReferencesForTable``` and redefinitions in dialects and subfunctions callstack. Because it looks like for MySQL dialect (that MariaDB uses in that case) ```getForeignKeyReferencesForTable``` in result calls ```getForeignKeysQuery(tableName, this.sequelize.config.database)``` from abstract dialect, but signature for MySQL dialect redefinition is ```getForeignKeysQuery(table, schemaName)```. And as a result, we have ```schemaName``` takes ```this.sequelize.config.database```. This is exactly what I wanted to show with my code. And I can't understand if there are some mistakes in my logic and I don't understand MySQL dialect or if there is an error in MySQL dialect code. This is just my helicopter view :) If you have any comments about this I will appreciate you:)

Reply:
I never used MariaDB, but https://mariadb.com/kb/en/schema/ said that schema and database are synonyms. Will prepare a new commit.